packages/testing/src/api/index.ts

```diff
@@ -64,6 +67,30 @@ export function initializeTestApp(options: any): admin.app.App {
   );
 }
 
+export function initializeFirestoreTestApp(options: any): FirebaseApp {
+  if (!('projectId' in options)) {
+    throw new Error('projectId not specified');
+  }
+  if (typeof options.auth != 'object') {
+    throw new Error('auth must be an object');
+  }
+  var header = {
+    alg: "RS256",
+    kid: "fakekid"
+  };
+  var fakeToken = [
+    util.base64.encodeString(JSON.stringify(header),/*webSafe=*/true),
+    util.base64.encodeString(JSON.stringify(options.auth),/*webSafe=*/true),
+    "fakesignature"
+  ].join(".");
+  return firebase.initializeApp({
+      projectId: options.projectId,
+      tokenOverride: fakeToken
+    },
+    'app-' + new Date().getTime() + "-" + Math.random()
+  );
```

Comment:
Can we initialize the app here, then immediately hijack the `INTERNAL.getToken` function?